### PR TITLE
Prevent duplicate tag push

### PR DESCRIPTION
**Pull request by scbedd**
...and ensure that ending up with a duplicate tag offers a seamless path forward.

Addresses #5712 

- [x] add integration test that actually exercises this scenario

This PR also updates some of the TestHelper infrastructure to:
- [x] Reduce clone time when cloning to a temp dir for pushing tag cleanup
- [x] Properly swap the `assets` config to a pushed integration tag (from C# integration tests). We were leaking tags from the C# integration tests. (Note the multiple additional `assets` cleanups)

### CODE REVIEW

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/Store/GitStore.cs [hunk 1]
```diff
@@ -124,7 +124,20 @@ public async Task<string> GetPath(string pathToAssetsJson)
...
+
+                    if (remoteResult.ExitCode != 0)
+                    {
+                        if (string.IsNullOrWhiteSpace(remoteResult.StdOut))
+                        {
+                            GitHandler.Run($"push origin {generatedTagName}", config);
+                        }
+                        else
+                        {
+                            _consoleWrapper.WriteLine("Not attempting to push new tag, as we have found it already within the assets repo");
```
Comment (benbp) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/Store/GitStore.cs:138]:
```suggestion
                            _consoleWrapper.WriteLine($"Not attempting to push tag '{generatedTagName}', as it already exists within the assets repo");
```

**Review by benbp — APPROVED** [review]

**Comment by scbedd** [comment]
Unfortunately I broke the integration tests somehow so this won't be going in as-is.

**Review by benbp — APPROVED** [review]

**Pull request merged**